agent/engine/docker_task_engine.go

```diff
@@ -1924,6 +1927,15 @@ func (engine *DockerTaskEngine) createContainer(task *apitask.Task, container *a
 		}
 	}
 
+	// If the container has mode=non-blocking set, and the max-buffer-size field is
+	// unset, then default to 10m.
+	if mode, ok := hostConfig.LogConfig.Config[logDriverMode]; ok {
+		_, hasBufferSize := hostConfig.LogConfig.Config[logDriverBufferSize]
+		if mode == string(dockercontainer.LogModeNonBlock) && !hasBufferSize {
```

Comment:
What will be the buffer size for blocking mode?

Reply:
In other words, what does the logging driver use as buffer size if no buffer size is provided?

Reply:
blocking mode does not have a buffer

Reply:
So every call is actually blocking? That's interesting. I'd be curious about its performance.

Reply:
yes, if the log driver has issues sending logs to the destination, the application faces backpressure: https://aws.amazon.com/blogs/containers/choosing-container-logging-options-to-avoid-backpressure/

Reply:
There's no configuration parameter for "blocking" mode, but I believe the log driver uses a buffered channel of a few hundred items, where each item is a separate log line.